Sources/ChessKit/Board.swift

```diff
@@ -198,13 +202,18 @@ public struct Board: Sendable {
 
     /// Determines end game state and
     /// handles pawn promotion for provided `move`.
-    private func process(move: Move) -> Move {
+    private mutating func process(move: Move) -> Move {
         var processedMove = move
 
         // checks & mate
         let checkState = self.checkState(for: move.piece.color)
         processedMove.checkState = checkState
-
+        
+        gamePositions.updateValue(
+            (gamePositions[position.hashed] ?? 0) + 1,
+            forKey: position.hashed
+        )
```

Comment:
If the other changes are applied this can become:
```suggestion
        // repetition
        positionHashCounts[position.hashValue, default: 0] += 1
```